### PR TITLE
feat(TCOMP-2226): Implement a default UI for streaming sources for user configuration of a StopStrategy

### PR DESCRIPTION
**Pull request by undx**
https://jira.talendforge.org/browse/TCOMP-2226

### CODE REVIEW

File: component-runtime-manager/src/main/java/org/talend/sdk/component/runtime/manager/ComponentManager.java [hunk 1]
```diff
@@ -1809,6 +1826,59 @@ public void onProcessor(final Class<?> type, final Processor processor) {
...
+                } else if (!root.getMetadata().containsKey(layoutType)) {
+                    root
+                            .getMetadata()
+                            .put(layoutType, layoutType.contains("gridlayout") ? layoutOptions : "true");
+                } else if (layoutType.contains("gridlayout")) {
+                    final String oldLayout = root.getMetadata().get(layoutType);
+                    root.getMetadata().put(layoutType, layoutOptions + "|" + oldLayout);
+                }
+            }
+            root.getNestedParameters().add(maxRecords);
```
Comment (ozhelezniak-talend) [inline component-runtime-manager/src/main/java/org/talend/sdk/component/runtime/manager/ComponentManager.java:1878]: layoutType == null
maxRecord & maxDurations go to "Advanced"
getNested -> "Main"

and at the end we add maxRecords & maxDuration into the "getNested"
(frankly the idea goes away from me with those layouts)

**Review by ozhelezniak-talend — APPROVED** [review]
LGTM, minor comments

File: component-runtime-manager/src/main/java/org/talend/sdk/component/runtime/manager/ComponentManager.java (hunk 1 above)
Comment (undx) [reply component-runtime-manager/src/main/java/org/talend/sdk/component/runtime/manager/ComponentManager.java:1869]: That's about default layout creation only. In all cases, maxR and maxD must be in nestedParameters.

File: component-runtime-impl/src/main/java/org/talend/sdk/component/runtime/input/Streaming.java [hunk 2]
```diff
@@ -84,23 +84,41 @@ public static RetryStrategy getStrategy(final LocalConfiguration configuration)
...
-        Long maxActiveTime =
-                ofNullable(System.getProperty(String.format("%s.talend.input.streaming.maxDurationMs", plugin)))
-                        .map(Long::parseLong)
-                        .orElse(ofNullable(configuration.get("talend.input.streaming.maxDurationMs"))
-                                .map(Long::parseLong)
-                                .orElse(null));
-        log.warn("[loadStopStrategy] {} {}", maxReadRecords, maxActiveTime);
+                ofNullable(internalConfiguration.entrySet()
+                        .stream()
+                        .filter(e -> e.getKey().startsWith("$maxRecords") || e.getKey().contains(".$maxRecords"))
```
[the remark this answers is not in the document]
Comment (undx) [reply component-runtime-impl/src/main/java/org/talend/sdk/component/runtime/input/Streaming.java]: [TCOMP-2237](https://jira.talendforge.org/browse/TCOMP-2237) Create a streaming configuration section for documentation

**Review by ozhelezniak-talend — APPROVED** [review]
lgtm

**Pull request merged**